### DISCUSSION TITLE
Example: how to repeat textures

### DISCUSSION THREAD

**Discussion by banesullivan** [pyvista/pyvista#2203] [Q&A]
I put up an [example on repeating textures](http://docs.vtki.org/examples/02-plot/plot-texture.html#repeating-textures) in the docs but it isn't a good implementation.

If you zoom in, the texture gets mapped funkily where the repetitions should intersect:

<img width="770" alt="Screen Shot 2019-04-08 at 1 48 06 PM" src="https://user-images.githubusercontent.com/22067021/55753009-8dfd4480-5a06-11e9-8a6b-d4fa415a2a2e.png">

Could someone help with correcting this example?

**Comment by GuillaumeFavelier** [comment]
I got it working by using the following in the example itself :

```py
axial_num_puppies = 4
xc = np.linspace(0, axial_num_puppies, curvsurf.dimensions[0])
yc = np.linspace(0, axial_num_puppies, curvsurf.dimensions[1])

xxc, yyc = np.meshgrid(xc, yc)
puppy_coords = np.c_[yyc.ravel(), xxc.ravel()]
```
I had to add `texture.RepeatOn()` in `pyvista/plotting.py` and remove the checking on the range for texture coordinates in `pyvista/common.py`. Maybe there is a better solution though.

This is what I obtained:
![image](https://user-images.githubusercontent.com/18143289/57850946-91db6e00-77df-11e9-81e2-2a7402e6f246.png)
